CONTRIBUTING.md

```diff
@@ -1,26 +1,126 @@
 # Developing hcloud-cli
 
 ## Generated files
+
 This repository contains generated files, mainly for testing purposes. These files are generated by running
+
 ```sh
 go generate ./...
 ```
+
 in the root directory of this repository. Make sure to keep generated files up-to-date
 when making changes to the code.
 
 ## Unit tests
-Unit tests are located in the `internal` directory. Run them with 
+
+Unit tests are located in the `internal` directory. Run them with
+
 ```sh
 go test ./...
 ```
 
 ## Build
-To build the binary, run 
+
+To build the binary, run
+
 ```sh
 go build -o hcloud-cli ./cmd/hcloud
 ```
 
-To include version information in the resulting binary and build for all targets, use GoReleaser: 
+To include version information in the resulting binary and build for all targets, use GoReleaser:
+
 ```sh
 goreleaser --snapshot --skip-publish --rm-dist
 ```
+
+## Conventions
+
+### Subcommand groups
+
+Cobra offers the functionality to group subcommands. The conventions on when and how to group commands are as follows:
+
+1. Use the following Categories:
+    - **General** (CRUD operations such as `create`, `delete`, `describe`, `list`, `update` + Label
+      commands `add-`/`remove-label`)
+    - Groups based on actions (e.g. `enable`, `disable`, `attach`, `detach`), for example `enable-`/`disable-protection`
+      or `poweron`/`poweroff`/`shutdown`/`reset`/`reboot`
+    - **Additional commands** that don't fit into the other categories (`Group.ID` is empty). These should be
+      utility commands, for example ones that perform client-side actions.
+2. Groups are only needed if more than the "General" group commands are present.
+3. `Group.ID` formatting:
+    1. If the `ID` is a noun, it should be singular.
+    2. If the `ID` is a verb, it should be in the infinitive form.
+    3. The `ID` should be in kebab-case.
+4. `Group.Title` formatting:
+    1. Should be the `ID` but capitalized and with spaces instead of dashes.
+    2. If a single resource is managed, the `Title` should be singular. Otherwise, it should be plural.
+
+       Example: Multiple network can be attached to server -> `Networks`, but only one ISO can be attached to a server -> `ISO`
+
+Here is how to create a group:
+
+```go
+cmd.AddGroup(&cobra.Group{ID: "general", Title: "General"})
```

Comment:
While testing this PR i noticed that all section headers in the help output end with a colon (`:`), while our custom groups do not have that colon. Do you think we should add it for consistency?

```shell
$ hcloud help
[...]

Usage:
[...]

Available Commands:
[...]

Flags:
[...]
```

Reply:
Done in af546fe55783a2b9c83775a7b598d4d1221e4657